lib/forklift/box_distributor.rb

```diff
@@ -0,0 +1,163 @@
+
+        machine.vm.hostname = if box.fetch('hostname', false)
+                                box.fetch('hostname')
+                              else
+                                "#{box.fetch('name').to_s.tr('.', '-')}.example.com"
+                              end
+
+        networks = configure_networks(box.fetch('networks', []))
+        configure_shell(machine, box)
+        configure_ansible(machine, box['ansible'], box['name'])
+        configure_libvirt(machie, box, networks)
+        configure_virtualbox(machine, box)
+        configure_rackspace(machine, box)
+
+        yield machine if block_given?
+      end
+    end
+
+    def configure_vagrant_hostmanager(config)
+      return unless Vagrant.has_plugin?('vagrant-hostmanager')
+      config.hostmanager.enabled = true
+      config.hostmanager.manage_host = true
+      config.hostmanager.manage_guest = true
+      config.hostmanager.include_offline = true
+    end
+
+    def configure_vagrant_cachier(config)
+      return unless Vagrant.has_plugin?('vagrant-cachier')
+      config.cache.scope = :box
+      config.cache.synced_folder_opts = {
+        type: :nfs,
+        mount_options: ['rw', 'vers=3', 'tcp', 'nolock']
+      }
+    end
+
+    def configure_networks(networks)
+      return if networks.empty?
+      networks.map do |network|
+        symbolized_options = network['options'].inject({}) { |memo, (k, v)| memo.update(k.to_sym => v) }
+        network.update('options' => symbolized_options)
+      end
+    end
+
+    def configure_ansible(machine, ansible, box_name)
+      return unless ansible
+      unless @ansible_groups[ansible['group'].to_s]
+        @ansible_groups[ansible['group'].to_s] = []
+      end
+
+      @ansible_groups[ansible['group'].to_s] << box.fetch('name')
+
+      if ansible.key?('server')
+        @ansible_groups["server-#{box_name}"] = ansible['server']
+      end
+
+      return unless (playbooks = ansible['playbook'])
+
+      [playbooks].flatten.each_with_index do |playbook, index|
+        args = SUPPORT_NAMED_PROVISIONERS ? ["main#{index}", type: 'ansible'] : [:ansible]
+        machine.vm.provision(*args) do |ansible_provisioner|
+          ansible_provisioner.playbook = playbook
+          ansible_provisioner.extra_vars = ansible['variables']
+          ansible_provisioner.groups = @ansible_groups
+        end
+      end
+    end
+
+    def configure_shell(machine, box)
+      return unless box.key?('shell') && !box['shell'].nil?
+      machine.vm.provision :shell do |shell|
+        shell.inline = box.fetch('shell')
+        shell.privileged = false if box.key?('privileged')
+      end
+    end
+
+    def configure_libvirt(machine, box, networks)
+      machine.vm.provider :libvirt do |p, override|
+        override.vm.box_url = box.fetch('libvirt') if box.fetch('libvirt', false)
+        override.vm.synced_folder '.', '/vagrant', type: 'rsync', rsync__args: ['--max-size=100m']
```

Comment:
I'd like this type to be easily overrideable. Especially with bats testing an nfs mount might be much nicer and those using cacher already have it set up.